docs/examples/llm_dataset_creation.rst

```diff
@@ -0,0 +1,110 @@
+Creating text dataset for LLM training using Lance
+---------------------------------------------------
+
+Lance can be used for creating and caching a text (or code) dataset for pre-training / fine-tuning of Large Language Models.
+The need for this arises when one needs to train a model on a subset of data or process the data in chunks without downloading
+all of it on the disk at once. This becomes a considerable problem when you just want a subset of a Terrabyte or Petabyte-scale dataset.
+
+In this example, we will be bypassing this problem by downloading a text dataset in parts, tokenizing it and saving it as a Lance dataset. 
+This can be done for as many or as few data samples as you wish with peak memory consumption never exceeding 3GBs!
+
```

Comment:
Should the 3GB number be "approx 3GB" depending on the device? I guess we don't want to promise a particular number or users might complain :)

Reply:
That does make sense